src/pgroonga.c

```diff
@@ -6095,6 +6103,17 @@ PGrnSearchBuildConditionBinaryOperation(PGrnSearchData *data,
 	PGrnExprAppendOp(data->expression, operator, 2, tag, NULL);
 }
 
+static void
+PGrnSearchSetMatchInStrategy(ScanKey key)
+{
+	// PostgreSQL 18 optimaize to "column &@ ANY (keyword1, keyword2, ...)"
+	// from "column &@ keyword1 OR column &@ keyword2 OR ...".
+	// &@ is match operator. So, we should handle
+	// "column &@ keyword1 OR column &@ keyword2 OR ..." as "match in"
+	// operator.
+	key->sk_strategy = PGrnMatchInStrategyV2Number;
```

Comment:
In general, we should not rewrite the given `ScanKey`... We should use it as a read-only data because it may be used by caller later.

Can we avoid rewriting the given `ScanKey`?